ektelo/workload.py

```diff
@@ -1,547 +1,201 @@
-from __future__ import division
-from builtins import str
-from builtins import zip
-from builtins import range
-import copy
-import hashlib
+from ektelo.matrix import EkteloMatrix, Identity, Ones, VStack, Kronecker
+import collections
 import itertools
-import numpy
-from ektelo.mixins import Marshallable
-from ektelo.mixins import CartesianInstantiable
-from ektelo.query_nd_union import ndRangeUnion
-from ektelo import util
+import numpy as np
 from scipy import sparse
-from scipy.sparse import linalg
-from functools import reduce
 
-"""
-These classes define workloads
-"""
-
-class Workload(Marshallable):
```

Comment:
If these classes are stateless (at least in the sense that their state can be regenerated from the initializer), then can we contemplate making them Marshallable as well? All it requires is to add `self.init_params = util.init_params_from_locals(locals())` to the top of the initializer.